src/molecules/Comp2Param.h

```diff
@@ -2,17 +2,22 @@
 #define COMP2PARAM_H_
 
 #include <vector>
+#include <memory>
 
 #include "molecules/Component.h"
 #include "molecules/Array2D.h"
 #include "molecules/ParaStrm.h"
+#include "molecules/mixingrules/MixingRuleBase.h"
 
 
 /** Comp2Param provides a flexible hash table interface to interaction parameter streams.
  *
  * @author Martin Bernreuther <bernreuther@hlrs.de> et al. (2010)
  */
 class Comp2Param {
+
+    using MixRulesType = std::map<int,std::map<int,std::shared_ptr<MixingRuleBase>>>;
```

Comment:
intuitively wouldn't it make more sense to move this to `MixingRuleBase` ? (not entirely sure...)

Reply:
I moved it to the `public` part of `MixingRuleBase`. I was not sure about the scoping but it works now.